src/Auth.ts

```diff
@@ -214,26 +219,35 @@ export class Auth {
       case AuthType.Browser:
         return this.getPublicClient(logger, debug);
       case AuthType.Certificate:
-        return this.getConfidentialClient(logger, debug, this.service.thumbprint as string, this.service.password);
+        return this.getConfidentialClient(logger, debug, this.service.thumbprint as string, this.service.password, undefined);
       case AuthType.Identity:
         // msal-node doesn't support managed identity so we need to do it manually
         return undefined;
+      case AuthType.Secret:
+        return this.getConfidentialClient(logger, debug, undefined, undefined, this.service.secret);
     }
   }
 
-  private getAuthClientConfiguration(logger: Logger, debug: boolean, certificateThumbprint?: string, certificatePrivateKey?: string): Msal.Configuration {
+  private getAuthClientConfiguration(logger: Logger, debug: boolean, certificateThumbprint?: string, certificatePrivateKey?: string, clientSecret?: string): Msal.Configuration {
     const msal: typeof Msal = require('@azure/msal-node');
     const { LogLevel } = msal;
     const cert = !certificateThumbprint ? undefined : {
       thumbprint: certificateThumbprint,
       privateKey: certificatePrivateKey as string
     };
+
+    const authConfig = cert ? {
+      clientId: this.service.appId,
+      authority: `https://login.microsoftonline.com/${this.service.tenant}`,
+      clientCertificate: cert
+    } : {
+      clientId: this.service.appId,
+      authority: `https://login.microsoftonline.com/${this.service.tenant}`,
+      clientSecret: clientSecret
+    };
+
```

Comment:
This works, but maybe we could use destructing here and avoid repeating `clientId` and `authority`.

```ts
const config = { 
      clientId: this.service.appId, 
      authority: `https://login.microsoftonline.com/${this.service.tenant}` 
};
    
const authConfig = cert 
      ? { ...config, clientCertificate: cert } 
      : { ...config, clientSecret };
```